project-base/storefront/components/Blocks/Popup/DeliveryAddressPopup.tsx

```diff
@@ -0,0 +1,237 @@
+
+    const [formProviderMethods] = useDeliveryAddressForm({
+        companyName: deliveryAddress?.companyName ?? '',
+        street: deliveryAddress?.street ?? '',
+        city: deliveryAddress?.city ?? '',
+        postcode: deliveryAddress?.postcode ?? '',
+        telephone: deliveryAddress?.telephone ?? '',
+        firstName: deliveryAddress?.firstName ?? '',
+        lastName: deliveryAddress?.lastName ?? '',
+        country: {
+            label: deliveryAddress?.country.name ?? '',
+            value: deliveryAddress?.country.code ?? '',
+        },
+    });
+    const formMeta = useDeliveryAddressFormMeta(formProviderMethods);
+    const countriesAsSelectOptions = useCountriesAsSelectOptions();
+    const { setValue } = formProviderMethods;
+
+    useEffect(() => {
+        if (countriesAsSelectOptions.length > 0 && deliveryAddress?.country) {
+            const selectedCountry = countriesAsSelectOptions.find(
+                (country) => country.value === deliveryAddress.country.code,
+            );
+            setValue(formMeta.fields.country.name, selectedCountry ? selectedCountry : countriesAsSelectOptions[0], {
+                shouldValidate: true,
+            });
+        }
+    }, [countriesAsSelectOptions, formMeta.fields.country.name, setValue]);
+
+    const deliveryAddressHandler: SubmitHandler<DeliveryAddressFormType> = async (deliveryAddressFormData) => {
+        blurInput();
+
+        if (deliveryAddress?.uuid) {
+            const editDeliveryAddressResult = await editDeliveryAddress({
+                input: {
+                    uuid: deliveryAddress.uuid,
+                    ...deliveryAddressFormData,
+                    country: deliveryAddressFormData.country.value,
+                },
+            });
+
+            updatePortalContent(null);
+
+            if (editDeliveryAddressResult.error !== undefined) {
+                showErrorMessage(
+                    t('There was an error while editing your delivery address'),
+                    GtmMessageOriginType.other,
+                );
+                return;
+            }
+
+            showSuccessMessage(t('Your delivery address has been edited'));
+        } else {
+            const createDeliveryAddressResult = await createDeliveryAddress({
+                input: {
+                    uuid: null,
+                    ...deliveryAddressFormData,
+                    country: deliveryAddressFormData.country.value,
+                },
+            });
+
+            updatePortalContent(null);
+
+            if (createDeliveryAddressResult.error !== undefined) {
+                showErrorMessage(
+                    t('There was an error while creating your delivery address'),
+                    GtmMessageOriginType.other,
+                );
+                return;
+            }
+
+            showSuccessMessage(t('Your delivery address has been created'));
+        }
```

Comment:
Since we do not have shared logic after this if statement I would use a guard clause to lower mental capacity and indentation to read and understand the code.

```ts
if (deliveryAddress?.uuid) {
    const editDeliveryAddressResult = await editDeliveryAddress({
        input: {
            uuid: deliveryAddress.uuid,
            ...deliveryAddressFormData,
            country: deliveryAddressFormData.country.value,
        },
    });
  
    updatePortalContent(null);
  
    if (editDeliveryAddressResult.error !== undefined) {
        showErrorMessage(
            t('There was an error while editing your delivery address'),
            GtmMessageOriginType.other,
        );
        return;
    }
  
    showSuccessMessage(t('Your delivery address has been edited'));
    return;
}

const createDeliveryAddressResult = await createDeliveryAddress({
    input: {
        uuid: null,
        ...deliveryAddressFormData,
        country: deliveryAddressFormData.country.value,
    },
});

updatePortalContent(null);

if (createDeliveryAddressResult.error !== undefined) {
    showErrorMessage(
        t('There was an error while creating your delivery address'),
        GtmMessageOriginType.other,
    );
    return;
}

showSuccessMessage(t('Your delivery address has been created'));
```